repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -2737,6 +2723,53 @@ public void repairAccesscontrolAlias(String guid) throws AtlasBaseException {
 
         RequestContext.get().endMetricRecord(metric);
     }
+
+    @Override
+    public void linkBusinessPolicy(String guid, List<String> linkGuids) throws AtlasBaseException {
+        processBusinessPolicy(guid, linkGuids, true);
+    }
+
+    @Override
+    public void unlinkBusinessPolicy(String guid, List<String> unlinkGuids) throws AtlasBaseException {
+        processBusinessPolicy(guid, unlinkGuids, false);
+    }
+
+    private void processBusinessPolicy(String guid, List<String> guids, boolean isLink) throws AtlasBaseException {
+        // Start recording the performance metrics for the operation
+        String operation = isLink ? "linkBusinessPolicy" : "unlinkBusinessPolicy";
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord(operation);
+
+        // Link or unlink the business policy to/from entities and retrieve the affected vertices
+        List<AtlasVertex> vertices = isLink ? this.entityGraphMapper.linkBusinessPolicy(guid, guids)
+                : this.entityGraphMapper.unlinkBusinessPolicy(guid, guids);
+
+        // If no vertices are returned, exit the method early
+        if (CollectionUtils.isEmpty(vertices)) {
+            return;
+        }
+
+        // Prepare the response for the entity mutations
+        EntityMutationResponse entityMutationResponse = new EntityMutationResponse();
+        for (AtlasVertex vertex : vertices) {
+            // Convert each vertex to an AtlasEntityHeader and add it to the response
+            AtlasEntityHeader entityHeader = entityRetriever.toAtlasEntityHeader(vertex);
+            entityMutationResponse.addEntity(UPDATE, entityHeader);
+        }
+
+        // Collect the vertices into a map for easier access by their GUID property
+        Map<String, AtlasVertex> vertexMap = vertices.stream()
```

Comment:
already iterating vertices above in for loop, populate this map in above for loop